spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/RunControllerHelperService.java

```diff
@@ -124,7 +123,7 @@ public JSONObject changePlatformType(HttpSession session, JSONObject json) {
         if (newPt != null) {
           log.info("STORED: " + newRuntype + " :: " + storedPlatformType);
           if (!newRuntype.equals(storedPlatformType)) {
-            run = dataObjectFactory.getRunOfType(newPt, user);
+            run = new RunImpl(user);
```

Comment:
Are the subclasses actually needed? The only difference between those classes is the constructor when passed the XML data from the client, which none of these methods do.